script.js

```diff
@@ -1,4 +1,348 @@
+//
+// MAKE DECK
+//
+//
+var makeDeck = function () {
+  var cardDeck = [];
+
+  // Initialise an array of the 4 suits in our deck. We will loop over this array.
+  var suits = ["hearts", "diamonds", "clubs", "spades"];
+
+  // Loop over the suits array
+  var suitIndex = 0;
+  while (suitIndex < suits.length) {
+    // Store the current suit in a variable
+    var currentSuit = suits[suitIndex];
+
+    // Loop from 1 to 13 to create all cards for a given suit
+    // Notice rankCounter starts at 1 and not 0, and ends at 13 and not 12.
+    // This is an example of a loop without an array.
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      // By default, the card name is the same as rankCounter
+      var cardName = rankCounter;
+      var cardValue = rankCounter;
+
+      if (cardName == 1) {
+        cardName = "ace";
+        cardValue = 1;
+      } else if (cardName == 11) {
+        cardName = "jack";
+        cardValue = 10;
+      } else if (cardName == 12) {
+        cardName = "queen";
+        cardValue = 10;
+      } else if (cardName == 13) {
+        cardName = "king";
+        cardValue = 10;
+      }
+
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        score: cardValue,
+      };
+
+      cardDeck.push(card);
+      rankCounter += 1;
+    }
+    suitIndex += 1;
+  }
+  return cardDeck;
+};
+//
+//
+//
+//
+//
+//
+//
+// SHUFFLE DECK
+//
+//
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Shuffle the elements in the cardDeck array
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+//
+//
+//
+//
+//
+//
+//
+//START MAIN FUNCTION
+//
+//
+var shuffledDeck = shuffleCards(makeDeck());
+var playerDeck = shuffledDeck.splice(0, 26);
+var computerDeck = shuffledDeck;
+var currentGameMode = "draw card";
+
+var playerCards = [];
+var computerCards = [];
+console.log("player cards: ");
+console.log(playerCards);
+console.log("computer cards: ");
+console.log(computerCards);
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  var playerScore = 0;
+  var computerScore = 0;
+
+  // Create a function to calcualte player's total score.
+  var calculatePlayerScore = function () {
```

Comment:
Interesting that you are creating the `calculatePlayerScore` function inside the `main` function.

Nothing wrong with that, but do remember rules of scoping. By creating `calculatePlayerScore` within `main`, you are scoping the variable `calculatePlayerScore` (which stores the definition of the function [from lines 116 to 129](https://github.com/rocketacademy/basics-blackjack/blob/a122772c1c05a29aae43234d1caabdbe74894d05/script.js#L116-L129)) within `main` alone.

This means you can only execute `calculatePlayerScore` like `calculatePlayerScore();` only in the `main` function. You wouldn't be able to execute it anywhere else outside of the `main` function, for instance in `shuffleCards` ([line 69 to 87](https://github.com/rocketacademy/basics-blackjack/blob/a122772c1c05a29aae43234d1caabdbe74894d05/script.js#L69-L87)) hypothetically. This is because `shuffleCards` is scoped globally, while `calculatePlayerScore` is scoped only to `main`.

Just wanted to ensure that you understand that by scoping `calculatePlayerScore` within `main`, you are also aware that you can only execute it within `main`. 🙂